### ISSUE TITLE
Cannot build Django project due to illegal character in path

### ISSUE THREAD

**Issue by vsfeedback** [microsoft/PTVS#6623]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Cannot-build-Django-project-due-to-illeg/1489556)._

---
[severity:I'm unable to use this version] [regression]
I am using VS19 in a Django project. Since the latest update the project doesn't build anymore due to this issue:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error		Cannot evaluate the item metadata "%(FullPath)". The item metadata "%(FullPath)" cannot be applied to the path "Debug mode: True
C:\Users\user\Development\Python\StableSwapper\StableSwapper\staticfiles\%2a%2a\%2a". Illegal characters in path.	StableSwapper	C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VisualStudio\v16.0\Python Tools\Microsoft.PythonTools.Django.targets	105	

```

I agree that `C:\.. \staticfiles\%2a%2a\%2a` looks pretty illegal. It seems that a sort of wildcard?

---
### Original Comments
(no comments)

---
### Original Solutions
(no solutions)

**Comment by greazer** [comment]
@linette-zyy, can you try and repro this in VS 2019 (16.10.4)? Thanks!

**Comment by linette-zyy** [comment]
@greazer 
I only get the VS2019 (16.10.5) from VS build and try to reproduce on this build. 
The issue does not provide which path he set, so I try to do the following steps.
Steps:
1. Create a Django project under this path: C:\Users\vyizh12\Development\Python\StableSwapper\StableSwapper\staticfiles\test\%2a%2a\%2a
2. Right click the Django project -->publish -->folder
3. Change Folder location to C:\Users\vyizh12\Development\Python\StableSwapper\StableSwapper\staticfiles\test\%2a%2a\%2a\DjangoWebProject1
![image](https://user-images.githubusercontent.com/59192491/127472372-240fb0de-9df8-4d97-ad51-af4fdb3e0a2d.png)
4. Build Django project
5. Observes

We got below error
![image](https://user-images.githubusercontent.com/59192491/127471557-290b8080-c3c4-486c-a44f-fbac528f3ac2.png)

If any others I can help with, please tell me.

**Comment by alvinsartor** [comment]
I'm the original poster. Let me give more information:

- The error is raised whenever I try to build the project (or run or do any action that triggers a build)

- That path does not exist: 
![image](https://user-images.githubusercontent.com/21270278/127630243-944ff9da-3c16-4b19-8748-faf33d564cb1.png)

- By doubleclicking on the error, I get to this part of code:
```
<ItemGroup>
      <_DjangoStaticFiles Include="$(DjangoStaticRootSetting)**\*" />
      <FilesForPackagingFromProject Include="@(_DjangoStaticFiles)" Condition="'%(FullPath)' != ''">
        <DestinationRelativePath>$([MSBuild]::MakeRelative($(QualifiedProjectHome), %(FullPath)))</DestinationRelativePath>
        <FromTarget>DjangoCollectStaticFiles</FromTarget>
        <Category>Run</Category>
      </FilesForPackagingFromProject>
    </ItemGroup>
```

- I've been running this project for 2 months and I've never seen this error before

**Comment by greazer** [comment]
Without access to the actual project, diagnosing this problem is rather difficult. Just doing an internet search I came across this tutorial that has a section that *may* relate to the problem you're having. Here's the link: https://www.kite.com/blog/python/django-static-files-templates-how-to/

The section I'm referring to says this: 

-----

## Django staticfiles app
### Configure staticfiles app in the project settings file
Before we get to any of the project’s frontend design, we need to change the default Django settings so that the `django.contrib.staticfiles` has the proper paths for where we want static files to live and be served from. Static files are what any CSS, JavaScript, and template files are called in Django.

Configuring the staticfiles app in Django gives us a couple things; Django will be able to collect static files from wherever they are in your project, including in apps that you install, into one location so that Django can then serve those files to the browser as they are requested. In addition to that, Django knows how to evaluate references to those paths within your templates, so in your templates, you can write {% static '/some/path' %} and that will work both in development and in production. This is because Django keeps track of where ‘static’ points to in the system even after static files are collected.

To configure the Django staticfiles app, we need to add STATICFILES_DIRS to our settings file. STATICFILES_DIRS is a `list` of all the places in which you have staticfiles. We just have one place, so we’ll just put that one place in the list. The directories in STATICFILES_DIRS must exist before collectstatic is run, as Django won’t create them for you. Then you need STATIC_ROOT which tells Django where to put the files after collection. This directory is created by Django based on the name you assign here, so you don’t have to create it beforehand.

-----

**Comment by alvinsartor** [comment]
Thanks for the answer @greazer. 
Unfortunately I cannot give free access to the project and could not create a minimum viable project with the same error. 

The section you pasted is about the initial configuration of the project, while I have been running this project for some months without problems and only had some after updating VS.

**Comment by alvinsartor** [comment]
I found the cause of the problem:

Sometimes the `pyproj` file contains a reference to a file that does not exist (i.e. it has been removed via a GIT GUI or VS had problems when it was deleted).
**Whenever this happens, the project won't compile and that will be the error displayed.**

Unfortunately finding the wrong line might be quite tedious as one has to check all of them manually.

Hope this will help someone.
